runtime/src/main/java/org/corfudb/runtime/view/AbstractOrderedGUIDFactory.java

```diff
@@ -0,0 +1,18 @@
+package org.corfudb.runtime.view;
+
+/**
+ * Globally Unique Identity (GUID) generator that returns compact GUIDs having a notion of
+ * a comparable global ordering.
+ *
+ * Why not use Java UUIDs?
+ *   JAVA UUIDs lack cross node comparability or global ordering in the absence of time sync.
+ *   They are also 36 bytes long which can be an overkill for simple use cases.
+ *
+ * What about schemes like Twitter Snowflake?
+ *    Requires time synchronization between cluster nodes for ordering.
+ *
+ * Created by Sundar Sridharan on 5/22/19.
+ */
+public interface AbstractOrderedGUIDFactory {
```

Comment:
The name Abstract* should be used for abstract class, right? But this is an interface.

Reply:
Any suggestions then?
(I wanted to avoid use of the I prefix even though it is used widely in the code today.)

Reply:
Maybe just remove Abstract to be OrderedGuidFactory?

Reply:
Just use `OrderedGuidFactory`
also, I don't see reasons for using interfaces for Factory classes

why you need an interface here? Maybe it's better to get rid of it and just use Util class with a static method

Reply:
Actually I can't make this a singleton because it is meant to be scoped to a corfu runtime. If we ever support multiple runtimes in the same jvm then they should have their own copies of the guid generators. If not, then it will be a singleton anyway.

Reply:
Need for interface is to have multiple types of ordered guid factories - one that generates snowflake style IDs and other which uses corfu's token based approach.